# Counting with featureCounts
rule counting:
    input:
        bams = expand("output/mapping/{srr}.bam", srr=SRR_LIST),
        gff  = "output/genome/reference.gff"
    output:
        "output/counts/all_samples_counts.txt"
    container:
        f"{ZENODO_URL}{SIF_LIST[1]}"
    params:
        outdir = "output/counts"
    threads: THREADS
    shell:
        """
        echo "Running featureCounts for all samples..."
        featureCounts -t gene -g ID -s 1 \
                      -a {input.gff} \
                      -o {output} \
                      {input.bams}
        echo "Combined featureCounts results saved to {output}"
        """
